scripts/run-on-vm.sh

```
@@ -414,6 +418,46 @@ if [[ "$RUN_PHASE" == "all" || "$RUN_PHASE" == "deploy" ]]; then
+        SYMLINK_PID=$!
+        echo "Symlink maintainer PID: $SYMLINK_PID"
+        cleanup_symlink() { [[ -n "$SYMLINK_PID" ]] && kill "$SYMLINK_PID" 2>/dev/null || true; }
+        trap cleanup_symlink EXIT
+    fi
+
+    # Start GNSS simulator as a pod for T-GM simulation tests
+    export GNSS_SIM_IMAGE="${IMG_PREFIX}:gnss-sim"
+    ./configGNSS.sh
+
+    # Read the gnss-sim pod's node IP for test framework API access
+    export GNSS_SIM_API_HOST="$(cat /tmp/gnss-sim-api-host 2>/dev/null || echo localhost)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)